rfcs/20200424-sparse-domain-isolation.md

```diff
@@ -0,0 +1,649 @@
+        'constant': lookup will return the valuse
+          filled by `initial_value`.
+      trainable: True, will be treated as a trainable Variable, and add to
+        to the list of variables collected in the graph under the key
+        `GraphKeys.TRAINABLE_VARIABLES`.
+      checkpoint: if True, the contents of the Variable are
+        saved to and restored from checkpoints.
+        If `shared_name` is empty for a checkpointed table,
+        it is shared using the table node name.
+
+    Returns:
+      A `dynamic_embedding.Variable` object.
+    """
+    self.key_dtype = key_dtype
+    self.value_dtype = value_dtype
+    self.default_value = initial_value
+    self.dim = dim
+    self.devices = devices if isinstance(devices, list) else [devices, ]
+    self.partition_fn = partitioner
+    self.name = name
+    self.shared_name = shared_name or "shared_name.{}".format(name)
+    self.initial_value = initial_value
+    self.initial_mode = initial_mode
+    self.use_default = self.initial_mode == 'constant'
+    self.trainable = trainable
+    self.checkpoint = checkpoint
+
+    self._tables = []
+    self.size_ops = []
+    self.shard_num = len(self.devices)
+
+    assert initial_mode in ["random", "constant"], \
+      "initial mode should be 'constant' or 'random' vs " + initial_mode
+
+    _default_value = _convert_anything_to_init(self.initial_value, self.dim)
+
+    for idx in range(len(self.devices)):
+      with ops.device(self.devices[idx]):
+        _mht = lookup_ops.MutableHashTable(key_dtype=self.key_dtype,
+                                           value_dtype=self.value_dtype,
+                                           default_value=_default_value,
+                                           use_default=self.use_default,
+                                           shared_name="{}-{}of{}".format(
+                                             self.shared_name,
+                                             idx + 1,
+                                             self.shard_num),
+                                           name="{}-{}of{}".format(self.name,
+                                                                   idx + 1,
+                                                                   self.shard_num),
+                                           checkpoint=self.checkpoint)
+
+        self._tables.append(_mht)
+        self.size_ops.append(self._tables[idx].size())
+
+  def upsert(self, keys, values, name=None):
+    """Insert or Update `keys` with `values`.
+
+    If key exists already, value will be updated.
+
+    Args:
+      keys: Keys to insert. Can be a tensor of any shape. Must match the table's
+        key type.
+      values: Values to be associated with keys. Must be a tensor of the same
+        shape as `keys` and match the table's value type.
+      name: A name for the operation (optional).
+
+    Returns:
+      The created Operation.
+    """
+    pass
+
+  def remove(self, keys, name=None):
```

Comment:
Could you add some details for when the remove would be triggered?

Reply:
For example, the users want to control the total number of features by some custom policy.

Reply:
Adding some docs would help.